sample-apps/react/messenger-clone/src/hooks/useCreateChatClient.ts

```diff
@@ -0,0 +1,51 @@
+import { useState, useEffect, useRef } from 'react';
+import {
+  ExtendableGenerics,
+  DefaultGenerics,
+  OwnUserResponse,
+  UserResponse,
+  TokenOrProvider,
+  StreamChat,
+} from 'stream-chat';
+
+export const useCreateChatClient = <
+  SCG extends ExtendableGenerics = DefaultGenerics,
+>({
+  apiKey,
+  userData,
+  tokenOrProvider,
+}: {
+  apiKey: string;
+  userData: OwnUserResponse<SCG> | UserResponse<SCG>;
+  tokenOrProvider: TokenOrProvider;
+}) => {
+  const [chatClient, setChatClient] = useState<StreamChat<SCG> | null>(null);
+  const disconnectRef = useRef(Promise.resolve());
```

Comment:
We should probably merge the PR in the Chat SDK and release it this week.

Reply:
@oliverlaz WDYT?

Reply:
Let's talk tomorrow about this.